source/server-discovery-and-monitoring/tests/integration/closeInUse-pool-clear.yml

```diff
@@ -0,0 +1,108 @@
+# Test connection pool clear(closeInUse:true).
+runOn:
+    # This test requires failCommand appName, and
+    # failCommand blockConnection with closeConnection:true (SERVER-53512).
+    - minServerVersion: "4.9"
+
+database_name: &database_name "sdam-tests"
+collection_name: &collection_name "closeInUse"
+
+data: []
+
+tests:
+  - description: pool clear close in use true generates expected events
+    useMultipleMongoses: false,
+    clientOptions:
+      connectTimeoutMS: 500, 
+      heartbeatFrequencyMS: 500
+      appname: "closeInUse"
+      retryReads: false
+      minPoolSize: 0
+    operations:
+      - name: insertMany
+        object: collection
+        arguments:
+          documents:
+            - _id: 1
+      - name: startThread
+        object: testRunner
+        arguments:
+          name: thread1
+      # simulate a long-running query
+      - name: runOnThread
+        object: testRunner
+        arguments:
+          name: thread1
+          operation:
+            name: find
+            object: collection
+            arguments:
+              filter: { "$where": "sleep(5000) || true" }
+      - name: configureFailPoint
+        object: testRunner
+        arguments:
+          failPoint:
+            configureFailPoint: failCommand
+            mode: { times: 1 }
+            data:
+                failCommands: ["hello"]
+                blockConnection: true
+                blockTimeMS: 1000
+                appName: closeInUse
+      - name: waitForEvent
+        object: testRunner
+        arguments:
+          event: ConnectionClosed
+          count: 1
+      - name: waitForEvent
+        object: testRunner
+        arguments:
+          event: PoolClearedEvent
+          count: 1
+      - name: waitForEvent
+        object: testRunner
+        arguments:
+          event: ConnectionCheckedInEvent
+          count: 2
+      - name: assertEventCount
+        object: testRunner
+        arguments:
+          event: PoolClearedEvent
+          count: 1
+      # ensure connection is closed
+      - name: assertEventCount
+        object: testRunner
+        arguments:
+          event: ConnectionClosed
+          count: 1 
+      # ensure there are corresponding checkOut/checkIn events
+      - name: assertEventCount
+        object: testRunner
+        arguments:
+          event: ConnectionCheckedOutEvent
+          count: 2
+      - name: assertEventCount
+        object: testRunner
+        arguments:
+          event: ConnectionCheckedInEvent
+          count: 2   
```

Comment:
I think we can drop these two checkedout/checkedin assertions, I think the others cover the situation adequately.

Reply:
nope, it's not covered. I mentioned it here https://github.com/mongodb/specifications/pull/1170#discussion_r953004623. In 2 words, at least for c# implementation, it's not possible to validate checkin events related to this PR in the pool test runner because the main checkin logic happens in error handling for level higher than the pool

Reply:
I meant that we can drop the `assertEventCount` operations on checkIn/checkOut since the above `waitForEvent` operation on `ConnectionCheckedInEvent` in this test seem adequate.

Reply:
ok, will remove it

Reply:
removed